test/WebJobs.Script.Tests/ContainerManagement/LinuxContainerInitializationHostServiceTests.cs

```diff
@@ -0,0 +1,145 @@
+                { EnvironmentSettingNames.ContainerStartContext, serializedContext },
+                { EnvironmentSettingNames.ContainerEncryptionKey, containerEncryptionKey },
+            };
+
+            // Enable Linux Container
+            AddLinuxContainerSettings(vars);
+
+            _instanceManagerMock.Setup(manager => manager.StartAssignment(It.Is<HostAssignmentContext>(context => hostAssignmentContext.Equals(context)))).Returns(true);
+
+            using (var env = new TestScopedEnvironmentVariable(vars))
+            {
+                _initializationHostService.StartAsync(CancellationToken.None).Wait();
+            }
+
+            _instanceManagerMock.Verify(manager => manager.StartAssignment(It.Is<HostAssignmentContext>(context => hostAssignmentContext.Equals(context))), Times.Once);
+        }
+
+        [Fact]
+        public void Assigns_Context_From_CONTAINER_START_CONTEXT_SAS_URI_If_CONTAINER_START_CONTEXT_Absent()
+        {
+            var containerEncryptionKey = TestHelpers.GenerateKeyHexString();
+            var hostAssignmentContext = GetHostAssignmentContext();
+            var encryptedHostAssignmentContext = GetEncryptedHostAssignmentContext(hostAssignmentContext, containerEncryptionKey);
+            var serializedContext = JsonConvert.SerializeObject(new { encryptedContext = encryptedHostAssignmentContext });
+
+            var vars = new Dictionary<string, string>
+            {
+                { EnvironmentSettingNames.ContainerStartContextSasUri, Containerstartcontexturi },
+                { EnvironmentSettingNames.ContainerEncryptionKey, containerEncryptionKey },
+            };
+
+            AddLinuxContainerSettings(vars);
+
+            var initializationHostService = new Mock<LinuxContainerInitializationHostService>(MockBehavior.Strict, new ScriptSettingsManager(), _instanceManagerMock.Object, NullLoggerFactory.Instance);
+
+            initializationHostService.Setup(service => service.Read(Containerstartcontexturi))
+                .Returns(Task.FromResult(serializedContext));
+
+            _instanceManagerMock.Setup(manager => manager.StartAssignment(It.Is<HostAssignmentContext>(context => hostAssignmentContext.Equals(context)))).Returns(true);
+
+            using (var env = new TestScopedEnvironmentVariable(vars))
+            {
+                initializationHostService.Object.StartAsync(CancellationToken.None).Wait();
+            }
+
+            _instanceManagerMock.Verify(manager => manager.StartAssignment(It.Is<HostAssignmentContext>(context => hostAssignmentContext.Equals(context))), Times.Once);
+        }
+
+        [Fact]
+        public void Does_Not_Assign_If_Context_Not_Available()
+        {
+            var vars = new Dictionary<string, string>();
+            AddLinuxContainerSettings(vars);
+
+            using (var env = new TestScopedEnvironmentVariable(vars))
+            {
+                _initializationHostService.StartAsync(CancellationToken.None).Wait();
```

Comment:
make this test async and remove the .Wait()

Reply:
Fixed